catalog/dags/common/elasticsearch.py

```diff
@@ -0,0 +1,223 @@
+            # the data available for search
+            refresh=True,
+            # Throttle
+            requests_per_second=requests_per_second,
+        )
+        return response["task"]
+
+    def _wait_for_reindex(task_id: str, expected_docs: int, es_host: str):
+        es_conn = ElasticsearchPythonHook(hosts=[es_host]).get_conn
+
+        response = es_conn.tasks.get(task_id=task_id)
+
+        count = response.get("task", {}).get("status", {}).get("total")
+        if expected_docs and count != expected_docs:
+            logger.info(
+                f"Reindexed {count} documents, but {expected_docs} were expected."
+            )
+        else:
+            logger.info(f"Reindexed {count} documents.")
+        return response.get("completed")
+
+    trigger_reindex_task = trigger_reindex(
+        es_host,
+        destination_index,
+        source_index,
+        query,
+        requests_per_second,
+        max_docs,
+        slices,
+    )
+
+    wait_for_reindex = PythonSensor(
+        task_id="wait_for_reindex",
+        python_callable=_wait_for_reindex,
+        timeout=timeout,
+        poke_interval=REFRESH_POKE_INTERVAL,
+        op_kwargs={
+            "task_id": trigger_reindex_task,
+            "expected_docs": max_docs,
+            "es_host": es_host,
+        },
+    )
+
+    trigger_reindex_task >> wait_for_reindex
+
+
+@task_group(group_id="point_alias")
+def point_alias(index_name: str, alias: str, es_host: str):
+    """
+    Point the target alias to the given index. If the alias is already being
+    used by one or more indices, it will first be removed from all of them.
+    """
+
+    @task.branch
+    def check_if_alias_exists(alias: str, es_host: str):
+        """Check if the alias already exists."""
+        es_conn = ElasticsearchPythonHook(hosts=[es_host]).get_conn
+        return (
+            "point_alias.remove_existing_alias"
+            if es_conn.indices.exists_alias(name=alias)
+            else "point_alias.point_new_alias"
+        )
+
+    @task
+    def remove_existing_alias(alias: str, es_host: str):
+        """Remove the given alias from any indices to which it  points."""
+        es_conn = ElasticsearchPythonHook(hosts=[es_host]).get_conn
+        response = es_conn.indices.delete_alias(
+            name=alias,
+            # Remove the alias from _all_ indices to which it currently
+            # applies
+            index="_all",
+        )
+        return response.get("acknowledged")
+
+    @task
+    def point_new_alias(
+        es_host: str,
+        index_name: str,
+        alias: str,
+    ):
+        es_conn = ElasticsearchPythonHook(hosts=[es_host]).get_conn
+        response = es_conn.indices.put_alias(index=index_name, name=alias)
+        return response.get("acknowledged")
+
+    exists_alias = check_if_alias_exists(alias, es_host)
+
+    remove_alias = remove_existing_alias.override(task_id="remove_existing_alias")(
```

Comment:
Why is `.override` necessary if it's directly reflecting the function name?

Reply:
Huh, zero reason -- I must have changed the function names at some point 😄 Cleaned up!